src/site/articles/mixins/index.markdown

```diff
@@ -248,30 +234,75 @@ in the underlying language.
 ### Statics
 
 Can one use the statics of the original class
-via the mixin application or not?  
+via the mixin application or not?
+
 Again, the answer (No) follows from the semantics of inheritance.
-Statics are not inherited in Dart. 
+Statics are not inherited in Dart.
 
 ### Types
 
-What is the type of a mixin application instance?
-In general, it is a subtype of its superclass,
-and also supports the methods defined on the mixin.
-The mixin name itself, however,
-denotes the type of the original class,
-which has its own superclass
-and may not be compatible with a particular mixin application.
-
-What about the interfaces a class supports?
-Does its mixin support them?
-In general, no, since interface support
-may rely on inherited functionality.
-This implies that a mixin application must declare
-what interfaces it implements explicitly.
+What is the type of a mixin application instance? In general, it is a
+subtype of its superclass, and also a subtype of the type denoted by
+the mixin name itself, i.e., the type of the original class.
 
-### Generics
+The original class has its own superclass. To ensure that a particular
+mixin application is compatible with the original class being mixed in,
+Dart places extra requirements on classes that use `with` clauses.
 
-If a class has type parameters,
-its mixin necessarily has identical type parameters. 
+If a class _A_ is defined using a `with` clause that applies a mixin _M_ where
+_M_ was derived from a class _K_, then _A_ must support the direct
+superinterfaces of _K_.
+
+{% prettyify dart %}
+class S {
+  twice(int x) => 2 * x;
+}
+
+abstract class I {
+   twice(x);
+}
+
+abstract class J {
+   thrice(x);
+}
+class K extends S implements I, J {
+  int thrice(x) => 3* x;
+}
+
+class B {
+  twice(x) => x + x;
+}
+class A = B with K;
+{% endprettify %}
+
+In particular, _A_ must support the implicit interface of the superclass
+_S_ of _K_.  This ensures that _A_ is indeed a subtype of _M_, even though
+it’s superclass chain is different. In our example above, _K_ needs to
+implement twice to meet the requirements of _I_ and must also implement
+thrice in order to satisfy the requirements imposed by _J_. _K_ meets these
```

Comment:
I usually like "()" after function/method names, but it seems especially important here, where "twice" and "thrice" aren't function-like names (they're adverbs, not nouns).

And/or consider using different names.